### DISCUSSION TITLE
Vec of remote type

### DISCUSSION THREAD

**Discussion by ohchase** [jonasbb/serde_with#506] [Q&A] [answered]
Hi, I'm struggling to figure out how to use this crate to deserialize a vec of a remote defined type.
Could anyone please provide some assistance; I believe this crate solves this use case but as of now I've been unable to achieve my target use case of deserializing a remote struct into a vec

I have found a similar discussion about this: https://github.com/jonasbb/serde_with/discussions/225 ; but still can't understand how to implement this for my use case. I apologize if this is considered a duplicate.

```Rust
#[derive(Serialize, Deserialize)]
#[serde(remote = "Vector3")]
struct Vector3Def {
    x: f32,
    y: f32,
    z: f32,
}

#[serde_as]
#[derive(Serialize, Deserialize, Debug)]
struct Route {
    #[serde_as(as = "Vec<Vector3Def>")]
    positions: Vec<Vector3>,
}

fn main() -> Result<()> {
    // Some JSON input data as a &str. Maybe this comes from the user.
    let data = r#"
        {
          "positions": [
            { "x":0, "y":0, "z:":0 },
            { "x":0, "y":0, "z:":0 },
            { "x":0, "y":0, "z:":0 },
            { "x":0, "y":0, "z:":0 },
          ]
        }"#;

    // Parse the string of data into serde_json::Value.
    let v: Route = serde_json::from_str(data)?;
    println!("{:#?}", v);

    Ok(())
}
```

**Comment by jonasbb** [comment, marked as answer]
The user guide in the documentation contains instructions how to combine the remote derive feature with `serde_as`:
https://docs.rs/serde_with/2.0.0/serde_with/guide/serde_as/index.html#using-serde_as-with-serdes-remote-derives

**Reply by ohchase** [reply]
Thank you very much, not sure why I somehow skipped over that section
